File: android/src/main/java/com/lightningkite/khrysalis/net/WebSocketFrame.actual.kt
package com.lightningkite.khrysalis.net

import com.lightningkite.khrysalis.bytes.Data
import io.reactivex.Observable
import okhttp3.Headers
import okhttp3.Request
import java.util.*

class WebSocketFrame(val binary: Data? = null, val text: String? = null) {
    override fun toString(): String {
        return text ?: binary?.let { "<Binary data length ${it.size}" } ?: "<Empty Frame>"
    }
}

File: ios/Khrysalis/src/net/WebSocketFrame.actual.swift
//Package: com.lightningkite.khrysalis.net
//Converted using Khrysalis2

import Foundation
import RxSwift
import RxRelay



public class WebSocketFrame: KStringable {
    
    public var binary: Data? 
    public var text: String? 
    
    
    public func toString() -> String {
        return text ?? { () in 
            if let it = (binary) {
                return "<Binary data length \(it.count)"
            }
            return nil
        }() ?? "<Empty Frame>"
    }
    
    public init(binary: Data?  = nil, text: String?  = nil) {
        self.binary = binary
        self.text = text
    }
    convenience public init(_ binary: Data? , _ text: String?  = nil) {
        self.init(binary: binary, text: text)
    }
}
 
